minard/RSTools.py

```diff
@@ -0,0 +1,478 @@
+# -*- coding: utf-8 -*-
+# Need comment above to use bullet-point character
+from wtforms import Form, BooleanField, StringField, PasswordField, validators
+from .db import engine, engine_nl
+from collections import OrderedDict
+import psycopg2
+import psycopg2.extras
+from dateutil import parser
+from .views import app
+
+def file_list_form_builder(formobj, runlists, data):
+    class FileListForm(Form):
+        pass
+
+    for (i, listname) in enumerate(runlists.keys()):
+        listnum = runlists[listname]
+        if listnum in data and formobj == -1:
+            setattr(FileListForm, '%s' % listname, BooleanField(label=listname, default='checked'))
+        else:
+            setattr(FileListForm, '%s' % listname, BooleanField(label=listname))
+
+        setattr(FileListForm, 'name', StringField('Name', [validators.Length(min=1), validators.InputRequired(), validators.Regexp('[A-Za-z0-9\s]{1,}', message='First and second name required.')]))
+        setattr(FileListForm, 'comment', StringField('Comment', [validators.InputRequired()]))
+        setattr(FileListForm, 'password', PasswordField('Password', [validators.InputRequired()]))
+
+    if formobj != -1:
+        return FileListForm(formobj)
+    else:
+        return FileListForm()
+
+def get_current_lists_run(run):
+    conn = engine.connect()
+    result = conn.execute("SELECT list FROM evaluated_runs WHERE run={}".format(run))
```

Comment:
@tannerbk @Jamicus96 to follow up on the discussion on the DWG call, this type of thing is not compatible with Python 2.6. When using `format`, you actually need to specify an index:
```python
"{}".format("test")  # Will fail with "ValueError: zero length field name in format"
"{0}".format("test")  # Will run
"{0} {1}".format("test1", "test2")  # Will run
```
You can also use keywords instead of indexes, so something like this can replace the highlighted code:
```python
"SELECT list FROM evaluated_runs WHERE run={run}".format(run=run)
```

Reply:
Ah thanks Mark, I looked at this, but didn't catch that difference. I'll fix these and redeploy.

Reply:
Ah I see, thanks Mark!